crates/noirc_frontend/src/monomorphisation/mod.rs

```diff
@@ -396,11 +396,13 @@ impl Monomorphiser {
             }
             HirPattern::Struct(_, patterns, _) => {
                 let fields = unwrap_struct_type(typ);
-                let patterns = patterns.into_iter().map(|(ident, pattern)| {
+                // We map each pattern to its respective field in a BTreeMap
+                // Fields in struct types are ordered, and doing this map guarantees we extract the correct field index
+                let patterns_map = btree_map(patterns, |(ident, pattern)| {
                     let typ = fields[&ident.0.contents].clone();
-                    (pattern, typ)
+                    (ident.0.contents, (pattern, typ))
                 });
-                self.unpack_tuple_pattern(value, patterns)
+                self.unpack_tuple_pattern(value, patterns_map.values().cloned())
```

Comment:
You're looking for `.into_values()`, no need to clone when you already own the BTreeMap

Reply:
Ah I didn't see method. Thank you, done